src/main/java/websocket/WSManager.java

```diff
@@ -119,6 +123,108 @@ public void sendRequest(Request request, int priority) throws ConnectException {
             }
         }
 
+        // Batch FileChangeRequests
+        if (request.data instanceof FileChangeRequest) {
+            // Add to batching queue
+            FileChangeRequest data = (FileChangeRequest) request.data;
+
+            if (!batchingByFile.containsKey(data.getFileID())) {
+                BatchingControl batchingControl = new BatchingControl();
+                batchingByFile.put(data.getFileID(), batchingControl);
+                batchingControl.maxVersionSeen = data.getBaseFileVersion();
+            }
+            BatchingControl batchingCtrl = batchingByFile.get(data.getFileID());
+
+            synchronized (batchingCtrl.patchBatchingQueue) {
+                batchingCtrl.patchBatchingQueue.addAll(Arrays.asList(((FileChangeRequest) request.data).getChanges()));
+            }
+
+            // Create releaser to make sure it only is ever done once
+            Runnable releaser = new Runnable() {
+                boolean released = false;
+
+                @Override
+                public void run() {
+                    synchronized (this) {
```

Comment:
Changed it just to be safe. I have no idea. Also, it's not a problem in any case, since nothing else uses it as a lock.